common/src/main/java/gov/cms/ab2d/common/service/CoverageServiceImpl.java

```diff
@@ -0,0 +1,258 @@
+
+    @Override
+    public CoveragePeriod getCoveragePeriod(long contractId, int month, int year) {
+        checkMonthAndYear(month, year);
+        return coveragePeriodRepo.getByContractIdAndMonthAndYear(contractId, month, year);
+    }
+
+    @Override
+    public boolean isCoveragePeriodInProgress(int searchId) {
+        CoveragePeriod period = coveragePeriodRepo.getOne(searchId);
+        JobStatus jobStatus = period.getStatus();
+        return jobStatus == JobStatus.IN_PROGRESS;
+    }
+
+    @Override
+    public boolean canEOBSearchBeStarted(int searchId) {
+        return !isCoveragePeriodInProgress(searchId);
+    }
+
+    @Override
+    public JobStatus getSearchStatus(int searchId) {
+
+        CoveragePeriod coverageSearch = coveragePeriodRepo.getOne(searchId);
+        return coverageSearch.getStatus();
+
+    }
+
+    @Override
+    public Optional<CoverageSearchEvent> findLastEvent(int searchId) {
+        CoveragePeriod period = coveragePeriodRepo.getOne(searchId);
+        return findLastEvent(period);
+    }
+
+    private Optional<CoverageSearchEvent> findLastEvent(CoveragePeriod period) {
+        return coverageSearchEventRepo.findFirstByCoveragePeriodOrderByCreatedDesc(period);
+    }
+
+    private CoverageSearchEvent getLastEvent(CoveragePeriod period) {
+        return findLastEvent(period).orElse(null);
+    }
+
+    @Transactional
+    @Override
+    public CoverageSearchEvent insertCoverage(int searchId, long searchEventId, Collection<String> beneficiaryIds) {
+
+        CoveragePeriod period = coveragePeriodRepo.getOne(searchId);
+        CoverageSearchEvent searchEvent = coverageSearchEventRepo.getOne(searchEventId);
+
+        Collection<Coverage> coverages = beneficiaryIds.stream()
+                .map(bene -> {
+                    Coverage coverage = new Coverage();
+                    coverage.setCoveragePeriod(period);
+                    coverage.setCoverageSearchEvent(searchEvent);
+                    coverage.setBeneficiaryId(bene);
+
+                    return coverage;
+                }).collect(toList());
+
+        JobStatus status = getSearchStatus(searchId);
+        if (status == JobStatus.IN_PROGRESS) {
+            coverageRepo.saveAll(coverages);
+        } else {
+            throw new InvalidJobAccessException("cannot update coverages if coverage search is not in progress (current status " + status + " )");
+        }
+
+        return searchEvent;
+    }
+
+    @Transactional
+    @Override
+    public void deletePreviousSearch(int searchId) {
+        CoveragePeriod period = coveragePeriodRepo.getOne(searchId);
+        Optional<CoverageSearchEvent> searchEvent = coverageSearchEventRepo.findPreviousSearch(searchId);
+
+        searchEvent.ifPresent(coverageSearchEvent ->
+                coverageRepo.removeAllByCoveragePeriodAndCoverageSearchEvent(period, coverageSearchEvent));
+    }
+
+    @Override
+    public CoverageSearchDiff searchDiff(int searchId) {
```

Comment:
Same here about where it's getting called.

Reply:
Not being called yet but eventually use to keep statistics about what was changed between searches.